### DISCUSSION TITLE
Curl is missing in the 2024-12-10 build

### DISCUSSION THREAD

**Discussion by Bartvz** [themactep/thingino-firmware#380] [General]
Since I am a serial updater of all my devices I ran `sysupgrade -p` on my Cinnado D1 t23n  which was on the 2024-12-10 build (BUILD_ID="master+2d66d9d, 2024-12-10 05:02:27 -0500") and got the output which I pasted below.
`root@Cinnado-D1 sbin# sysupgrade -p
Run as /sbin/sysupgrade -p
Updating self
/sbin/sysupgrade: line 69: curl: not found
chmod: /tmp/sysupgrade/sysupgrade: No such file or directory
/sbin/sysupgrade: line 69: curl: not found
Re-running script from /tmp/sysupgrade
/sbin/sysupgrade: exec: line 248: /tmp/sysupgrade/sysupgrade: not found`

Curl seems to be missing. wget is included but won't download from https.

**Comment by davlaw** [comment]
Was coming to look up this issue myself for a wyze v2 that I mess up the other day and nothing would work to recover it.  So I cracked it open and did the clone tool on it and did get it recovered. Then discovered this same issue.  Guess I will be following this thread for best recover\y options as Thingino is new to me, have only discovered it a few days ago ;)   

Added: did try to put a bin file on the sdcard, but curl is still needed



NAME=Thingino
ID=thingino
VERSION="1 (Ciao)"
VERSION_ID=1
VERSION_CODENAME=ciao
PRETTY_NAME="Thingino 1 (Ciao)"
ID_LIKE=buildroot
CPE_NAME="cpe:/o:thinginoproject:thingino:1"
LOGO=thingino-logo-icon
ANSI_COLOR="1;34"
HOME_URL="https://thingino.com/"
ARCHITECTURE=mips
IMAGE_ID=wyze_c2_jxf23
BUILD_ID="master+2d66d9d, 2024-12-10 05:02:27 -0500"
HOSTNAME=ing-wyze-c2
TIME_STAMP=1733706349
BUILDROOT_NAME=Buildroot
BUILDROOT_VERSION=-g791cb854
BUILDROOT_ID=buildroot
BUILDROOT_VERSION_ID=2025.02-git
BUILDROOT_PRETTY_NAME="Buildroot 2025.02-git"
 
Run as /sbin/sysupgrade -p
Creating /tmp/sysupgrade directory
Updating self
/sbin/sysupgrade: line 69: curl: not found
chmod: /tmp/sysupgrade/sysupgrade: No such file or directory
/sbin/sysupgrade: line 69: curl: not found
Re-running script from /tmp/sysupgrade
/sbin/sysupgrade: exec: line 248: /tmp/sysupgrade/sysupgrade: not found

**Comment by gtxaspec** [comment]
Yep, sorry, there was an issue with the build on that date.  Recommend you download the firmware from the github releases, and put it on the sd card, as `autoupdate-full.bin` this will flash the latest full firmware to your device, then you can set it up again.

**Comment by davlaw** [comment]
Thanks, that was easy enough.  Thinino is a recent adoption in my arsenal, thanks for the help with a "noob" question

**Comment by themactep** [comment]
[curl.gz](https://github.com/user-attachments/files/18139930/curl.gz)

1. download
2. gunzip curl.gz
3. scp -O curl root@cameraip:/bin/
4. ssh root@cameraip chmod +x /bin/curl

Profit!

**Reply by Bartvz** [reply]
> [curl.gz](https://github.com/user-attachments/files/18139930/curl.gz)
> 
>     1. download
> 
>     2. gunzip curl.gz
> 
>     3. scp -O curl root@cameraip:/bin/
> 
>     4. ssh root@cameraip chmod +x /bin/curl
> 
> 
> Profit!

Worked like a charm, thank you!
Saves me the "hassle" of preparing an SD card and redoing settings.

**Comment by gtxaspec** [comment]
moving to discussions for visibility as this is already fixed
